### ISSUE TITLE
Download Manager

### ISSUE THREAD

**Issue by vext01** [browsh-org/browsh#126]
Hi,

Cool project.

I thought I could use browsh to download things on my server from google drive (which needs authentication). However, the download box never appears. I assume it isn't supported, but that would be super useful.

Thanks

**Comment by tobimensch** [comment]
Should be possible to implement such a feature.

**Comment by tombh** [comment]
AFAIK, the download should still happen, it's just that you can't see the dialogue box or any progress. Is the file not in `~/Downloads`?

**Comment by vext01** [comment]
Ah, I had no idea. I ended up using VNC to download the files on the server. It was pretty clumsy and browsh would have been better :)

**Comment by tombh** [comment]
Ha, never mind. I should add it to the documentation.

**Comment by vext01** [comment]
Or better, extend the user interface to have a download manager.

**Comment by tombh** [comment]
Ah yes, good point!

Though I currently have no idea if the Webextension or the RDP has anyway to detect downloads...

**Comment by vext01** [comment]
(btw, is there any chance you could put some build instructions in `README.md`. I'd be tempted to port this to OpenBSD (although I have no idea if it would work. We have go and firefox at least). Shall I open an issue?

**Comment by vext01** [comment]
Glad to hear that it works.

There's not much point in distributing binaries for OpenBSD though as there's no guarantee of backward compat. It'd be better to provide some detailed build instructions and have someone at OpenBSD build a port.

**Comment by tombh** [comment]
How do you mean that there's no guarantee of backward compatibility? As in, building for OpenBSD versions today are guaranteed to work in newer versions of OpenBSD?

**Comment by vext01** [comment]
OpenBSD freely changes its ABI whenever it feels the need to, so a binary for release X.Y will probably not work on X.Y+1. Lots of people don't run releases either. It's not going to be practical to distribute binaries like that.

If we get an OpenBSD port in their tree, then packages get build alongside the OS and thus will always work at any given time.

I'm an OpenBSD developer and can help with the port, but I would need detailed build instructions.

Thanks.

**Comment by tombh** [comment]
Understood. I'm going to rewrite the build scripts to refactor out the webextension build a signing, which will make it possible to self-build the entire binary with the only caveat being having to download the webextension zip file and bundle it as a binary blob. Watch #143

**Comment by gvanem** [comment]
PS. I tried rewriting some of `build_browsh.sh` into a Windows `get-web-extension.bat` file. It fails on the command:
```
curl -L -o browsh.xpi https://github.com/browsh-org/browsh/releases/download/v1.4.10/browsh 1.4.10-an.fx.xpi
```
(does it look okay?) The content of `browsh.xpi` contains this mumbo-jumbo:
```
<Error><Code>InvalidArgument</Code>
<Message>Only one auth mechanism allowed; only the X-Amz-Algorithm query parameter, 
Signature query string parameter or the Authorization header should be specified</Message><ArgumentName>Authorization</ArgumentName>
<ArgumentValue>Basic YW5vbnltb3VzOnlvdXJAZW1haWwuYWRkcmVzcw==</ArgumentValue>...
```

Seems related to the fact I *do have* a line for `github.com` in my `%APPDATA%/_netrc` file.
But simply using Wget, the download worked!

**Comment by tombh** [comment]
I just realised this isn't the right issue to be discussing this. Sorry to any future readers of this thread for the digression!

So I'll just answer quickly here, then if you have any other problems unrelated to the original title, just open up a new issue :)

I think you just missed out a hyphen in the URL, so this should work: `curl -L -o browsh.xpi https://github.com/browsh-org/browsh/releases/download/v1.4.10/browsh-1.4.10-an.fx.xpi`

**Comment by gvanem** [comment]
No, that was just a typo. The mumbo-jumbo remains. But dropping my `_netrc` credential, it works:
```
curl -L -o browsh.xpi -u ':' https://github.com/browsh-org/browsh/releases/download/v1.4.10/browsh-1.4.10-an.fx.xpi
```

(569.148 bytes). But yikes, I think I'll stick to Wget.

**Comment by tombh** [comment]
Oh I see. Ok, great that it works now.
